akka-docs/src/main/paradox/typed/persistence-snapshot.md

```diff
@@ -55,10 +57,69 @@ an actor tries to store a snapshot, at which point the operation will fail.
-restart.
+the `SnapshotCompleted` or `SnapshotFailed` signal. Snapshot failures are, by default, logged but do not cause
+the actor to stop or restart.
 
 If there is a problem with recovering the state of the actor from the journal when the actor is
-started, `onRecoveryFailure` is called (logging the error by default), and the actor will be stopped.
+started, `RecoveryFailed` signal is emitted (logging the error by default), and the actor will be stopped.
 Note that failure to load snapshot is also treated like this, but you can disable loading of snapshots
 if you for example know that serialization format has changed in an incompatible way.
+
+## Snapshot deletion
+
+To free up space, an event sourced actor can automatically delete older snapshots based on the given `RetentionCriteria`.
+
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #retentionCriteria }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #retentionCriteria #snapshottingPredicate }
+
+Snapshot deletion is triggered after saving a new snapshot.
+
+The above example will save snapshots automatically every `numberOfEvents = 100`. Snapshots that have sequence
+number less than the sequence number of the saved snapshot minus `keepNSnapshots * numberOfEvents` (`100 * 2`) are automatically
+deleted.
+
+In addition, it will also save a snapshot when the persisted event is `BookingCompleted`. Automatic snapshotting
+based on `numberOfEvents` can be used without specifying @scala[`snapshotWhen`]@java[`shouldSnapshot`]. Snapshots
+triggered by the @scala[`snapshotWhen`]@java[`shouldSnapshot`] predicate will not trigger deletion of old snapshots.
+
+On async deletion, either a `DeleteSnapshotsCompleted` or `DeleteSnapshotsFailed` signal is emitted.
+You can react to signal outcomes by using @scala[with `receiveSignal` handler] @java[by overriding `receiveSignal`].
+By default, successful completion is logged by the system at log level `debug`, failures at log level `warning`.
+
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #retentionCriteriaWithSignals }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #retentionCriteriaWithSignals }
+
+## Event deletion
+
+Deleting events in event sourcing based applications is typically either not used at all, or used in conjunction with snapshotting.
+If snapshot-based retention is enabled, after a snapshot has been successfully stored, a delete of the events
+(journaled by a single event sourced actor) up until the sequence number of the data held by that snapshot can be issued.
```

Comment:
Maybe add the caveat from the end in this section also, that deleting events makes it impossible to add new state-logic and replay events. I think this is a bit too unopinionated, would be better somewhat recommending against deleting events.